mlflow/utils/request_utils.py

```diff
@@ -187,7 +187,27 @@ def _get_http_response_with_retries(
     session = _get_request_session(
         max_retries, backoff_factor, backoff_jitter, retry_codes, raise_on_status
     )
-    return session.request(method, url, **kwargs)
+
+    # the environment variable is hardcoded here to avoid importing mlflow.
+    # however, documentation is available in environment_variables.py
+    env_value = os.getenv("MLFLOW_DISABLE_HTTP_REDIRECTS", "0").lower()
+    disable_redirects = env_value in ["true", "1"]
+
+    if disable_redirects:
+        kwargs["allow_redirects"] = False
+
+    response = session.request(method, url, **kwargs)
+    response_is_redirect = response.is_redirect or 300 <= response.status_code < 400
+
+    if disable_redirects and response_is_redirect:
```

Comment:
```suggestion
    if disable_redirects and (response.is_redirect or 300 <= response.status_code < 400):
```

nit (not a blocker): to avoid useless evaluation